web-app/django/VIM/apps/instruments/views/instrument_detail.py

```diff
@@ -0,0 +1,45 @@
+from django.views.generic import DetailView
+from VIM.apps.instruments.models import Instrument, Language
+
+
+class InstrumentDetail(DetailView):
+    """
+    Displays details of a specific instrument.
+    """
+
+    model = Instrument
+    template_name = "instruments/detail.html"
+    context_object_name = "instrument"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        instrument = self.get_object()
+
+        context["wikidata_id"] = instrument.wikidata_id
+        context["default_image_url"] = (
+            instrument.default_image.url if instrument.default_image else None
+        )
+        context["hornbostel_sachs_class"] = instrument.hornbostel_sachs_class
+        context["mimo_class"] = instrument.mimo_class
+
+        # Get english name
+        context["instrument_name_en"] = instrument.instrumentname_set.get(
+            language__en_label="english"
+        ).name

```

Comment:
You'll get the english instrument name below when you loop through all the names and you could extract it into a separate context item there, so again, this leads to 1 extra SQL query.